lib/src/gotrue_client.dart

```diff
@@ -44,6 +44,8 @@ class GoTrueClient {
   int _refreshTokenRetryCount = 0;
 
   final _onAuthStateChangeController = BehaviorSubject<AuthState>();
+  final _onAuthStateChangeControllerSync =
+      BehaviorSubject<AuthState>(sync: true);
```

Comment:
This is great! I genuinely did not know that you can just create synchronous behavior subject like this. I know you asked this question somewhere, but I wonder if it will cause any issues if we just update the existing `_onAuthStateChangeController` to synchronous. I feel synchronous is more of the expected behavior, and we probably could consider it a bug fix.

Reply:
From the [SynchronousStreamController](https://api.dart.dev/dev/2.8.0-dev.7.0/dart-async/SynchronousStreamController-class.html) documentation.
> The synchronous broadcast stream controller also has a restrictions that a normal stream controller does not: The [add](https://api.dart.dev/dev/2.8.0-dev.7.0/dart-async/SynchronousStreamController/add.html), [addError](https://api.dart.dev/dev/2.8.0-dev.7.0/dart-async/SynchronousStreamController/addError.html), [close](https://api.dart.dev/dev/2.8.0-dev.7.0/dart-async/SynchronousStreamController/close.html) and [addStream](https://api.dart.dev/dev/2.8.0-dev.7.0/dart-async/StreamController/addStream.html) methods must not be called while an event is being delivered. That is, if a callback on a subscription on the controller's stream causes a call to any of the functions above, the call will fail. A broadcast stream may have more than one listener, and if an event is [add](https://api.dart.dev/dev/2.8.0-dev.7.0/dart-async/SynchronousStreamController/add.html)ed synchronously while another is being also in the process of being added, the latter event might reach some listeners before the former. To prevent that, an event cannot be added while a previous event is being fired. This guarantees that an event is fully delivered when the first add, [addError](https://api.dart.dev/dev/2.8.0-dev.7.0/dart-async/SynchronousStreamController/addError.html) or [close](https://api.dart.dev/dev/2.8.0-dev.7.0/dart-async/SynchronousStreamController/close.html) returns, and further events will be delivered in the correct order.

I don't think we can ensure that. It feels you have to be careful what you do with a sync stream and feel more safe if we use it internally only. From a quick search, flutterfire auth doesn't use a sync stream as well, but that might not even be possible, because of their native binding.

Reply:
@Vinzent03 
Thanks for this. Sounds good!